Aptoide/aptoide-client-v8
# [DT-763] Fix OOS dialog space calculation

**What does this PR do?**

This PR aims at fixing a bug where when the app is uninstalled, the size on the out of space dialog wouldnt get updated.
This implied a kind of big refactor of the way we were handling the uninstallation. I added mroe details on the slack channel message about the decisions, which will be further doccumentated on appropriate platforms.

**Database changed?**

No

**Where should the reviewer start?**

- [ ] AvailableSpaceViewModel.kt

**How should this be manually tested?**

Get a device with low storage (or make it by installing several apps), try to install an app that doesnt fit the device and check if the flow works ok.

**What are the relevant tickets?**

  Tickets related to this pull-request: [DT-763](https://aptoide.atlassian.net/browse/DT-763)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 



**Code Review Checklist**
